erdos/src/node/leader.rs

```diff
@@ -221,6 +301,181 @@ impl Leader {
 
         let mut id_of_this_worker = WorkerId::default();
 
+        fn handle_messages_from_worker(
+            msg_from_worker: WorkerNotification,
+            id_of_this_worker: WorkerId,
+            channel_to_leader: &UnboundedSender<WorkerHandlerNotification>,
+        ) {
+            match msg_from_worker {
+                WorkerNotification::JobUpdate(job_graph_id, job, job_state) => {
+                    tracing::trace!(
+                        "[WorkerHandler {}] Job {:?} from JobGraph {:?} is in state {:?}.",
+                        id_of_this_worker,
+                        job,
+                        job_graph_id,
+                        job_state,
+                    );
+                    if let Err(error) = channel_to_leader.send(
+                        WorkerHandlerNotification::JobUpdate(job_graph_id, job, job_state.clone()),
+                    ) {
+                        tracing::warn!(
```

Comment:
Hmmm, the idea here was to reserve `Error` for serious errors after which the execution may be seriously impaired (as is discussed in the [tracing API](https://docs.rs/tracing-core/0.1.5/src/tracing_core/metadata.rs.html#241-262)). The reason why this was left as a `warn` was so that we can catch if these hazardous situations occur, and develop retry techniques if needed (e.g, by caching messages, reconstructing channels and retrying the `send`)

Reply:
Alright, then I'm ok with keeping it as `warn`.